### ISSUE TITLE
Different order of options since I updated to 1.5.4

### ISSUE THREAD

**Issue by uli-heller** [nextcloud/polls#1160]
### What is going wrong?
**Describe the bug**
I created a poll with these options:

- 3
- 4
- 5
- ...
- 12

With the old version of "polls", they show up in chronological (=numerical) ordering.
I created them one after the other, so I see 3,4,5,...,12

After updating to 1.5.4, the ordering is different:
![image](https://user-images.githubusercontent.com/1677414/96006863-35e6a000-0e3e-11eb-9ae0-44b59dd63d22.png)

The option 6 shows up in a wrong place.

**To Reproduce**
Steps to reproduce the behavior (what I did):

1. Create a poll with polls-1.4.3 with the options 1,2,3,4,5,6,7,8
2. Ask some people to participate
3. Delete options 1,2
4. Add options 9,10,11,12
5. Verify: The options show up in this ordering: 3,4,5,6,7,8,9,10,11,12
6. Update to polls-1.5.4
7. Error: The options show up in this ordering: 3,6,4,5,7,8,9,10,11,12

**Expected behavior**
Keep the ordering after updating to 1.5.4: 3,4,5,6,7,8,9,10,11,12

**Screenshots**

![image](https://user-images.githubusercontent.com/1677414/96007479-cde48980-0e3e-11eb-82cb-7d0b7c27a29a.png)

### Information about your polls installation
**Polls version?** (see apps page) 1.5.4

**Fresh installation or update from a prior version (from which one)?** Update - 1.4.3

**How did you install this version?(Appstore or describe installation)** Download the tar.gz, unpack

### Information about your Instance of Nextcloud/ownCloud
**Nextcloud or ownCloud?** Nextcloud

**Nextcloud/ownCloud version:** (see Nextcloud admin page) 20.0.0

**List of activated apps:**
```
If you have access to your command line run e.g.:
sudo -u www-data php occ app:list
from within your Nextcloud installation folder
Enabled:
  - accessibility: 1.6.0
  - activity: 2.13.1
  - admin_audit: 1.10.0
  - calendar: 2.1.2
  - cloud_federation_api: 1.3.0
  - comments: 1.10.0
  - contactsinteraction: 1.1.0
  - dashboard: 7.0.0
  - dav: 1.16.0
  - deck: 1.1.1
  - documentserver_community: 0.1.8
  - drawio: 0.9.7
  - federatedfilesharing: 1.10.1
  - federation: 1.10.1
  - files: 1.15.0
  - files_external: 1.11.1
  - files_pdfviewer: 2.0.1
  - files_rightclick: 0.17.0
  - files_sharing: 1.12.0
  - files_trashbin: 1.10.1
  - files_versions: 1.13.0
  - files_videoplayer: 1.9.0
  - firstrunwizard: 2.9.0
  - forms: 2.0.4
  - keeweb: 0.6.3
  - logreader: 2.5.0
  - lookup_server_connector: 1.8.0
  - nextcloud_announcements: 1.9.0
  - notes: 4.0.0
  - notifications: 2.8.0
  - oauth2: 1.8.0
  - onlyoffice: 6.0.2
  - password_policy: 1.10.1
  - photos: 1.2.0
  - polls: 1.5.4
  - privacy: 1.4.0
  - provisioning_api: 1.10.0
  - recommendations: 0.8.0
  - serverinfo: 1.10.0
  - settings: 1.2.0
  - sharebymail: 1.10.0
  - spreed: 10.0.0
  - support: 1.3.0
  - survey_client: 1.8.0
  - systemtags: 1.10.0
  - tasks: 0.13.4
  - text: 3.1.0
  - theming: 1.11.0
  - twofactor_backupcodes: 1.9.0
  - updatenotification: 1.10.0
  - user_ldap: 1.10.2
  - user_status: 1.0.0
  - viewer: 1.4.0
  - weather_status: 1.0.0
  - workflowengine: 2.2.0
Disabled:
  - encryption
  - mail
```

**Nextcloud configuration:**
```
If you have access to your command line run e.g.:
sudo -u www-data php occ config:list system
from within your Nextcloud installation folder

or

Insert your config.php content here
Make sure to remove all sensitive content such as passwords. (e.g. database password, passwordsalt, secret, smtp password, …)

{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_proxies": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "internal.daemons-point.com",
            "83.236.132.110",
            "10.2.100.166",
            "192.168.0.199",
            "192.168.0.195"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "overwritehost": "internal.daemons-point.com",
        "overwriteprotocol": "https",
        "overwritewebroot": "\/nextcloud",
        "overwrite.cli.url": "https:\/\/internal.daemons-point.com\/nextcloud",
        "dbtype": "mysql",
        "version": "20.0.0.9",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "apps_paths": [
            {
                "path": "\/var\/www\/nextcloud\/apps",
                "url": "\/apps",
                "writable": false
            },
            {
                "path": "\/var\/www\/nextcloud\/apps-local",
                "url": "\/apps-local",
                "writable": false
            }
        ],
        "loglevel": 2,
        "theme": "",
        "maintenance": false,
        "trashbin_retention_obligation": "auto",
        "appstoreenabled": false,
        "htaccess.RewriteBase": "\/nextcloud",
        "memcache.local": "\\OC\\Memcache\\APCu",
        "ldapIgnoreNamingRules": false,
        "ldapProviderFactory": "\\OCA\\User_LDAP\\LDAPProviderFactory",
        "mail_smtpmode": "smtp",
        "mail_smtpsecure": "tls",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "587",
        "mail_smtpauth": 1,
        "mail_smtpauthtype": "PLAIN",
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "mysql.utf8mb4": true,
        "mail_sendmailmode": "smtp",
        "app_install_overwrite": [
            "drawio",
            "keeweb",
            "documentserver_community",
            "polls"
        ]
    }
}

```

### Server configuration

**Operating system**:
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=18.04
DISTRIB_CODENAME=bionic
DISTRIB_DESCRIPTION="Ubuntu 18.04.5 LTS"

**Web server:**
apache2                                                  2.4.29-1ubuntu4.14

**Database:**
mariadb-server                                           1:10.1.44-0ubuntu0.18.04.1

**PHP version:**
php7.2                                                   7.2.24-0ubuntu0.18.04.6

**Signing status:**
```
Login as admin user into your Nextcloud and access
http://example.com/index.php/settings/integrity/failed
paste the results here.

No errors have been found.
```

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
LDAP

**Comment by dartcafe** [comment]
Nice find, thanks for reporting.

This bug occurs in 1.4. When adding new options, the order was not calculated properly. In your case the orders where build in the following way (Option text:Option order): 
- After Step 1: 1:1,2:2,3:3,4:4,5:5,6:6,7:7,8:8
- After Step 3: 3:3,4:4,5:5,6:6,7:7,8:8
- After Step 4: 3: 3,4:4,5:5,6:6,7:7,8:8,9:**7**,10:**8**,11:**9**,12:**10**  

The options where just counted and new options would get count + 1 as order. Since 1.5 new options get the highest order number + 1.

But I will check this again, to make sure, the bug is eliminted. I have the feeling, that this could be hardened.

**Comment by uli-heller** [comment]
Thx. Which table and which column contains the ordering? Knowing these 
would allow me to fix the ordering of my existing poll. Thanks for 
helping, Uli

Am 2020-10-15 08:13, schrieb René Gieling:

> Nice find, thanks for reporting.
> 
> This bug occurs in 1.4. When adding new options, the order was not 
> calculated properly. In your case the orders where build in the 
> following way (Option text:Option order):
> 
> * After Step 1: 1:1,2:2,3:3,4:4,5:5,6:6,7:7,8:8
> * After Step 3: 3:3,4:4,5:5,6:6,7:7,8:8
> * After Step 4: 3: 3,4:4,5:5,6:6,7:7,8:8,9:7,10:8,11:9,12:10
> 
> The options where just counted and new options would get count + 1 as 
> order. Since 1.5 new options get the highest order number + 1.
> 
> But I will check this again, to make sure, the bug is eliminted. I have 
> the feeling, that this could be hardened.
> 
> --
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub [1], or unsubscribe 
> [2].


Links:
------
[1] 
https://github.com/nextcloud/polls/issues/1160#issuecomment-708923091
[2]

**Comment by dartcafe** [comment]
polls_options. But it should be simply achievable by just ordering the option in the sidebar.

**Comment by uli-heller** [comment]
Sorry for asking before researching:

```

MariaDB [owncloud]> select * from oc_polls_options;
+-----+---------+-------------------------------------------------------------------------------------------------------------------------------------+------------+------------+-----------+
| id | poll_id | poll_option_text | timestamp | order | confirmed |
+-----+---------+-------------------------------------------------------------------------------------------------------------------------------------+------------+------------+-----------+

...

| 150 | 14 | 3 | 0 | 0 | 0 |
| 151 | 14 | 4 | 0 | 0 | 0 |
| 152 | 14 | 5 | 0 | 0 | 0 |
| 153 | 14 | 6 | 0 | 0 | 0 |
| 154 | 14 | 7 | 0 | 0 | 0 |

| 194 | 14 | 8 | 0 | 6 | 0 |
| 195 | 14 | 9 | 0 | 7 | 0 |
| 196 | 14 | 10 | 0 | 8 | 0 |
| 197 | 14 | 11 | 0 | 9 | 0 |
| 198 | 14 | 12 | 0 | 10 | 0 |

```

Seems easy to fix for this concrete poll...

Am 2020-10-15 08:34, schrieb Uli Heller:

> Thx. Which table and which column contains the ordering? Knowing these 
> would allow me to fix the ordering of my existing poll. Thanks for 
> helping, Uli
> 
> Am 2020-10-15 08:13, schrieb René Gieling:
> 
>> Nice find, thanks for reporting.
>> 
>> This bug occurs in 1.4. When adding new options, the order was not 
>> calculated properly. In your case the orders where build in the 
>> following way (Option text:Option order):
>> 
>> * After Step 1: 1:1,2:2,3:3,4:4,5:5,6:6,7:7,8:8
>> * After Step 3: 3:3,4:4,5:5,6:6,7:7,8:8
>> * After Step 4: 3: 3,4:4,5:5,6:6,7:7,8:8,9:7,10:8,11:9,12:10
>> 
>> The options where just counted and new options would get count + 1 as 
>> order. Since 1.5 new options get the highest order number + 1.
>> 
>> But I will check this again, to make sure, the bug is eliminted. I 
>> have the feeling, that this could be hardened.
>> 
>> --
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub [1], or unsubscribe 
>> [2].


Links:
------
[1] 
https://github.com/nextcloud/polls/issues/1160#issuecomment-708923091
[2]

**Comment by uli-heller** [comment]
I tried the sidebar. Unfortunately, it doesn't work. Dragging the "6" 
downwards works at first, but when releasing the mouse it flips back to 
its original position. I'm using chrome 86. Shall I open another issue 
for this?

Am 2020-10-15 08:43, schrieb René Gieling:

> polls_options. But it should be simply achievable by just ordering the 
> option in the sidebar.
> 
> --
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub [1], or unsubscribe 
> [2].


Links:
------
[1] 
https://github.com/nextcloud/polls/issues/1160#issuecomment-708936083
[2]

**Comment by dartcafe** [comment]
Waaaah. Yes. There is a bug in the ordering function. Has nothing to do with upgrade process. A new issue would be good.

**Comment by uli-heller** [comment]
I created #1163
